[NA] Fill in required-settings and features before create

alexander-cit opened the pull request:
If you add a new item of required-settings or a new feature, you don't need to update seeds and tests anymore for it to work. That's what DEFAULT means :smile: 

The previous attempt was here https://github.com/CitizenLabDotCo/citizenlab/pull/3291
But now, after this PR https://github.com/CitizenLabDotCo/citizenlab/pull/4983, we won't need to update pricing plans (no need to run rake tasks) in more cases ([like here](https://github.com/CitizenLabDotCo/citizenlab/pull/4732/files#diff-f1922f7571df32ef9822ca67a0c95052b6017690b7fee5f42c002c7d228c75d2R761)).

alexander-cit committed:
[NA] Fill in required-settings and features before create

If you add a new item of required-settings or a new feature,
you don't need to update seeds anymore for it to work.

alexander-cit committed:
[CL-3466] Use allowed/enabled defaults from schema

Cherry-picked from https://github.com/CitizenLabDotCo/citizenlab/pull/4983

alexander-cit committed:
[CL-3466] Simplify setting default allowed/enabled

Cherry-picked from https://github.com/CitizenLabDotCo/citizenlab/pull/4732/commits

alexander-cit commented:
@adessy https://github.com/CitizenLabDotCo/citizenlab/pull/3291#issuecomment-1325079533

> Hmm, I'm not sure about that one. I'm a bit concerned this could hide bugs in tenant creation. For instance, if we try to create a tenant with a partial settings object, this would set default values to the missing feature flags/settings that would not necessarily match our pricing plans.

Do you mean the case when we create a tenant on AdminHQ? In this case, the form is populated with the default values from `back/config/schemas/settings.schema.json.erb` anyway. And then these default values are sent from AdminHQ to CL.

@sebastienhoorens What do you think about this PR?

sebastienhoorens reviewed (approved):
Looks go to me :)

adessy reviewed (approved):
I was not sure to agree back then ; I do agree today :) Let's go for it!

alexander-cit committed:
[CL-1483] Do not allow/enable FC/CU login methods by default

alexander-cit committed:
[CL-1483] Fix specs after add_missing_features_and_settings, on: :create

alexander-cit committed:
[CL-1483] Create user that is confirmed by default in specs

Previously, the user_confirmation feature was disabled by default
in specs (but not in production). It means that
user#confirmation_required? returns false.

In this PR, we enabled it by default in specs, and so
user#confirmation_required? started to return true.

And now we need create users that are confirmed by default.

alexander-cit committed:
[CL-1483] Disable user_confirmation in specs to avoid fixing all tests

alexander-cit committed:
[CL-1483] Disable verification in specs to avoid fixing all tests
